src/poetry/core/packages/package.py

```diff
@@ -215,11 +217,13 @@ def _set_version(
         from poetry.core.constraints.version import Version
 
         if not isinstance(version, Version):
-            self._version = Version.parse(version)
-            self._pretty_version = pretty_version or version
-        else:
-            self._version = version
-            self._pretty_version = pretty_version or self._version.text
+            try:
+                version = Version.parse(version)
+            except InvalidVersion:
+                raise ValueError(f"Invalid version '{version}' on package {self.name}")
```

Comment:
Why a `ValueError` and not a more specific exception like `InvalidVersion` with an updated error message?

Reply:
can't say I feel very strongly about it